File: prms-web/src/app/pages/project/services/project-member.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '../../../core/services/base-service';
import { ProjectMemberWritePayload } from '../models/project-member.model';
import { ProjectMemberSearchRequest } from '../models/project-member-search.request';

@Injectable({
  providedIn: 'root',
})
export class ProjectMemberService {
  private readonly apiUrl = '/api/v1/project-members';

  constructor(private base: BaseService) {}

  getPage(request: ProjectMemberSearchRequest) {
    return this.base.post(this.apiUrl + '/page', request);
  }

  getById(id: string) {
    return this.base.get(`${this.apiUrl}/${id}`);
  }

  create(request: ProjectMemberWritePayload) {
    return this.base.post(this.apiUrl, request);
  }

  update(id: string, request: ProjectMemberWritePayload) {
    return this.base.put(`${this.apiUrl}/${id}`, request);
  }

  delete(id: string) {
    return this.base.delete(`${this.apiUrl}/${id}`);
  }
}

File: src/main/java/com/tranhuudat/prms/service/ProjectMemberService.java
package com.tranhuudat.prms.service;

import com.tranhuudat.prms.dto.BaseResponse;
import com.tranhuudat.prms.dto.project_member.ProjectMemberDto;
import com.tranhuudat.prms.dto.project_member.ProjectMemberSearchRequest;

import java.util.UUID;

public interface ProjectMemberService {

    BaseResponse create(ProjectMemberDto request);

    BaseResponse update(UUID id, ProjectMemberDto request);

    BaseResponse delete(UUID id);

    BaseResponse getById(UUID id);

    BaseResponse getPage(ProjectMemberSearchRequest request);
}
